src/libraries/System.IO.Compression/tests/ZipArchive/zip_InvalidParametersAndStrangeFiles.cs

```diff
@@ -907,6 +907,382 @@ public static async Task ZipArchive_InvalidHuffmanData()
+
+                        Assert.Equal(10, bytesRead);
+
+                        Assert.Equal(new byte[] { 0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09 }, uncompressedBytes);
+                    }
+                }
+
+                updatedStream.Seek(0, SeekOrigin.Begin);
+
+                // Create a new entry, forcing the central directory headers to be rewritten. The local file header
+                // for first.bin would normally be skipped (because it hasn't changed) but it needs to be rewritten
+                // because the central directory headers will be rewritten with a valid value and the local file header
+                // needs to match.
+                using (ZipArchive updatedArchive = new ZipArchive(updatedStream, ZipArchiveMode.Update, leaveOpen: true))
+                {
+                    ZipArchiveEntry newEntry = updatedArchive.CreateEntry("second.bin", CompressionLevel.NoCompression);
+
+                    // Add data to the new entry
+                    using (Stream entryStream = newEntry.Open())
+                    {
+                        entryStream.Write(new byte[] { 0x00, 0x01, 0x02, 0x03 });
+                    }
+                }
+
+                byte[] updatedContents = updatedStream.ToArray();
+                // Verify that the local file header and the central directory headers have both been rewritten, and both have
+                // the correct value.
+                int updatedLocalVersionToExtract = updatedContents[4];
+                int updatedCentralDirectoryVersionToExtract = updatedContents[101];
+
+                Assert.Equal(20, updatedCentralDirectoryVersionToExtract);
+                Assert.Equal(20, updatedLocalVersionToExtract);
+
+                updatedStream.Seek(0, SeekOrigin.Begin);
+                // Following an update of the ZipArchive, reopen it in read-only mode. Make sure that both entries are correct.
+
+                using (ZipArchive updatedArchive = new ZipArchive(updatedStream, ZipArchiveMode.Read, true))
+                {
+                    Assert.Equal(2, updatedArchive.Entries.Count);
+
+                    ZipArchiveEntry firstEntry = updatedArchive.Entries[0];
+                    ZipArchiveEntry secondEntry = updatedArchive.Entries[1];
+
+                    Assert.Equal("first.bin", firstEntry.Name);
+                    Assert.Equal(10, firstEntry.Length);
+
+                    Assert.Equal("second.bin", secondEntry.Name);
+                    Assert.Equal(4, secondEntry.Length);
+
+                    using (Stream entryStream = firstEntry.Open())
+                    {
+                        byte[] uncompressedBytes = new byte[firstEntry.Length];
+                        int bytesRead = entryStream.Read(uncompressedBytes);
+
+                        Assert.Equal(10, bytesRead);
+
+                        Assert.Equal(new byte[] { 0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09 }, uncompressedBytes);
```

Comment:
This expected array is used in the 3 new tests, consider making it a private field that the 3 tests can consume.